app/services/notifications/notifiable_action/send.rb

```diff
@@ -16,16 +16,16 @@ def self.call(...)
       end
 
       def call
+        return unless notifiable.is_a?(Article)
+
         json_data = {
           user: user_data(notifiable.user),
           article: article_data(notifiable)
         }
         json_data[:organization] = organization_data(notifiable.organization) if notifiable.organization_id
 
         notifications_attributes = []
-        # followers is an array and not an activerecord object
-        # followers can occasionally be nil because orphaned follows can possibly exist in the db (for now)
-        followers.sort_by(&:updated_at).reverse[0..10_000].each do |follower|
+        notifiable.followers.sort_by(&:updated_at).last(10_000).reverse_each do |follower|
```

Comment:
Nice! 🙌🏼

I'm surprised that we don't have any logic to limit the number of followers this pulls back. The `users` table has 138 columns in it so each row is pretty big. Do you know what endpoint or Sidekiq job invokes this? I'd like to get some metrics on the number of results.

I'm not suggesting fixing it in this PR, to be clear. It's been like this, so it can stay for a while longer and I'll probably cap it myself in a future PR.

Reply:
`Notifications::NotifiableAction::Send` is invoked by [Notifications::NotifiableActionWorker](https://github.com/forem/forem/blob/14416f79cc02f002e34f53339ed28a27adee1a48/app/workers/notifications/notifiable_action_worker.rb#L11) which is invoked by [Notification.send_to_followers](https://github.com/forem/forem/blob/14416f79cc02f002e34f53339ed28a27adee1a48/app/models/notification.rb#L44) which is invoked by:

https://github.com/forem/forem/blob/14416f79cc02f002e34f53339ed28a27adee1a48/app/services/articles/updater.rb#L30-L32

https://github.com/forem/forem/blob/14416f79cc02f002e34f53339ed28a27adee1a48/app/services/articles/creator.rb#L21

which are used in all "create or update" operations in articles. During create the notifications are sent if the article has the published flag set to true and during updated the notifications are sent if the article's status goes from "draft" to "published"

Reply:
Whoops, sorry @jgaskins, I completely missed your comment/question from yesterday!  @rhymes has got it right -- the Creator + Updater services call on the `send_to_followers` method, which kicks of a worker, which calls on NotifiableAction::Send.

Spoiler alert: I will be touching all of this code as a part of [RFC #22](https://github.com/forem/rfcs/pull/22) 😂